### PR TITLE
Adding more details to the average_probabilities.csv file for classification tasks

### PR DESCRIPTION
**Pull request by orhunguley**
This PR will add more details to "average_probabilities.csv" file, which should be created after running inference for classification models. An example file is attached.[averaged_probabilities.csv](https://github.com/CBICA/GaNDLF/files/8189415/averaged_probabilities.csv)

## Proposed Changes

- Added headers to the file, "SubjectID", "PredictedClass", "ClassName"

## Checklist

- [ x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x ] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [ x] Non-breaking change (would not break existing functionality): please provide as many details as possible for any breaking change 
- [x ] Function/class source code documentation added/updated
- [x ] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [ x] If applicable, version information [has been updated in GANDLF/version.py]
- [x ] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [x ] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [ x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation

### CODE REVIEW

File: GANDLF/compute/forward_pass.py [hunk 1]
```diff
@@ -412,13 +416,23 @@ def validate_network(
...
+            logits_df[class_list] = logit_tensor
+
+            logits_df.to_csv(
+                os.path.join(current_fold_dir, "logits.csv"), index=False, sep=","
             )
+            # np.savetxt(
+            #     os.path.join(current_fold_dir, "logits.csv"),
+            #     logit_tensor.detach().cpu().numpy(),
+            #     delimiter=",",
+            # )
```
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py:435]:
```suggestion
```

File: GANDLF/inference_manager.py [hunk 2]
```diff
@@ -51,16 +52,30 @@ def InferenceManager(dataframe, outputDir, parameters, device):
...
+                # fold_logits = np.genfromtxt(logits_dir, delimiter=",")
+                class_list = [str(c) for c in parameters["model"]["class_list"]]
+                fold_logits = pd.read_csv(logits_dir)[class_list].values
                 fold_logits = torch.from_numpy(fold_logits)
                 fold_probs = F.softmax(fold_logits, dim=1)
                 probs_list.append(fold_probs)
 
     if probs_list and is_classification:
+        columns = ["SubjectID", "PredictedClass"] + parameters["model"]["class_list"]
+        # columns = ["SubjectID",] + parameters["model"]["class_list"]
```
Comment (sarthakpati) [inline GANDLF/inference_manager.py:64]:
```suggestion
```

File: GANDLF/inference_manager.py [hunk 3]
```diff
@@ -51,16 +52,30 @@ def InferenceManager(dataframe, outputDir, parameters, device):
...
+            class_list[a] for a in averaged_probs.argmax(1)
+        ]
+        averaged_probs_df.to_csv(
+            os.path.join(outputDir, "averaged_probabilities.csv"), index=False, sep=","
         )
+        # np.savetxt(
+        #     os.path.join(outputDir, "averaged_probabilities.csv"),
+        #     averaged_probs,
+        #     delimiter=",",
+        # )
```
Comment (sarthakpati) [inline GANDLF/inference_manager.py:81]:
```suggestion
```

**Review by sarthakpati — CHANGES_REQUESTED** [review]
Looks good, just need to remove the commented out lines since they appear to be redundant.

**Review by orhunguley — COMMENTED** [review]
removed the commented out lines.

File: GANDLF/inference_manager.py [hunk 4]
```diff
@@ -51,16 +52,24 @@ def InferenceManager(dataframe, outputDir, parameters, device):
...
-        np.savetxt(
-            os.path.join(outputDir, "averaged_probabilities.csv"),
-            averaged_probs,
-            delimiter=",",
+        averaged_probs_df[class_list] = averaged_probs
+        averaged_probs_df.PredictedClass = [
+            class_list[a] for a in averaged_probs.argmax(1)
+        ]
+        averaged_probs_df.to_csv(
+            os.path.join(outputDir, "averaged_probabilities.csv"), index=False, sep=","
```
Comment (sarthakpati) [inline GANDLF/inference_manager.py:74]:
```suggestion
            os.path.join(outputDir, "final_predictions_with_averaged_probabilities.csv"), index=False, sep=","
```

**Review by sarthakpati — APPROVED** [review]
LGTM!

**Comment by sarthakpati** [comment]
This is merged.

**Pull request closed without merging**